sentry_sdk/integrations/litestar.py

```diff
@@ -0,0 +1,282 @@
+
+    def injection_wrapper(self: "Litestar", *args: "Any", **kwargs: "Any") -> None:
+        after_exception = kwargs.pop("after_exception", [])
+        kwargs.update(
+            after_exception=[
+                exception_handler,
+                *(
+                    after_exception
+                    if isinstance(after_exception, list)
+                    else [after_exception]
+                ),
+            ]
+        )
+
+        SentryLitestarASGIMiddleware.__call__ = SentryLitestarASGIMiddleware._run_asgi3  # type: ignore
+        middleware = kwargs.pop("middleware", None) or []
+        kwargs["middleware"] = [SentryLitestarASGIMiddleware, *middleware]
+        old__init__(self, *args, **kwargs)
+
+    Litestar.__init__ = injection_wrapper
+
+
+def patch_middlewares() -> None:
+    old__resolve_middleware_stack = BaseRouteHandler.resolve_middleware
+
+    def resolve_middleware_wrapper(self: "Any") -> "List[Middleware]":
+        return [
+            enable_span_for_middleware(middleware)
+            for middleware in old__resolve_middleware_stack(self)
+        ]
+
+    BaseRouteHandler.resolve_middleware = resolve_middleware_wrapper
+
+
+def enable_span_for_middleware(middleware: "Middleware") -> "Middleware":
+    if (
+        not hasattr(middleware, "__call__")  # noqa: B004
+        or middleware is SentryLitestarASGIMiddleware
+    ):
+        return middleware
+
+    if isinstance(middleware, DefineMiddleware):
+        old_call: "ASGIApp" = middleware.middleware.__call__
+    else:
+        old_call = middleware.__call__
+
+    async def _create_span_call(
+        self: "MiddlewareProtocol",
+        scope: "LitestarScope",
+        receive: "Receive",
+        send: "Send",
+    ) -> None:
+        if sentry_sdk.get_client().get_integration(LitestarIntegration) is None:
+            return await old_call(self, scope, receive, send)
+
+        middleware_name = self.__class__.__name__
+        with sentry_sdk.start_span(
+            op=OP.MIDDLEWARE_LITESTAR,
+            description=middleware_name,
+            origin=LitestarIntegration.origin,
+        ) as middleware_span:
+            middleware_span.set_tag("litestar.middleware_name", middleware_name)
+
+            # Creating spans for the "receive" callback
+            async def _sentry_receive(
+                *args: "Any", **kwargs: "Any"
+            ) -> "Union[HTTPReceiveMessage, WebSocketReceiveMessage]":
+                with sentry_sdk.start_span(
+                    op=OP.MIDDLEWARE_LITESTAR_RECEIVE,
+                    description=getattr(receive, "__qualname__", str(receive)),
+                    origin=LitestarIntegration.origin,
+                ) as span:
+                    span.set_tag("litestar.middleware_name", middleware_name)
+                    return await receive(*args, **kwargs)
+
+            receive_name = getattr(receive, "__name__", str(receive))
+            receive_patched = receive_name == "_sentry_receive"
```

Comment:
This logic will fail to detect Sentry patching after adding the `@ensure_integration_enabled` decorator. Likely the best solution will be to modify `@ensure_integration_enabled` to set a `_sentry_patched` attribute on everything it patches. I can prepare a PR to do this

Reply:
Should I wait until you make that PR to make the `@ensure_integration_enabled` changes to avoid those failures?

Reply:
I decided to go ahead and make all the @ensure_integration_enabled changes except one that results in several of the tests failing. Once the PR you mentioned is in, I can add that one as well.

Reply:
Yes, sounds good! I will only get around to making this PR tomorrow at the earliest

Reply:
Now that we are removing the `ensure_integration_enabled` decorator from these functions, this logic will continue to work. I still think it might be helpful to set a `_sentry_patched` attribute on everything `@ensure_integration_enabled` patches, but this PR no longer depends on that change.